internal/controller/operatorconfigmap_controller.go

```diff
@@ -153,6 +154,28 @@ func (c *OperatorConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		),
 	)
 
+	nodeSpecificLabelChangePredicate := builder.WithPredicates(
+		predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				oldLabels := e.ObjectOld.GetLabels()
+				newLabels := e.ObjectNew.GetLabels()
+
+				keysToWatch := []string{
+					"topology.rook.io/rack",
+					"topology.kubernetes.io/zone",
+					"kubernetes.io/hostname",
+				}
+
+				for _, key := range keysToWatch {
+					if oldLabels[key] != newLabels[key] {
+						return true
+					}
+				}
+				return false
+			},
+		},
+	)
```

Comment:
DeleteFunc, CreateFunc, and GenericFunc defaults to true and will trigger the reconciliation of the controller

Reply:
Used generic predicate